src/Config.php

```diff
@@ -40,14 +41,24 @@ final class Config
      * @param string $rootPath The path to the project root where composer.json is located.
      * @param string|null $configsPath The path to where configs are stored.
      * @param string|null $environment The environment name.
+     * @param array $mergeRecursive Names of config files that should be merged recursively.
      *
      * @throws ErrorException If the environment does not exist.
      */
-    public function __construct(string $rootPath, string $configsPath = null, string $environment = null)
+    public function __construct(string $rootPath, string $configsPath = null, string $environment = null, array $mergeRecursive = [])
     {
         $this->rootPath = $rootPath;
         $this->relativeConfigsPath = trim($configsPath ?? Options::DEFAULT_CONFIGS_DIRECTORY, '/');
         $this->configsPath = $this->rootPath . '/' . $this->relativeConfigsPath;
+        $this->mergeRecursive = array_merge(
+            $mergeRecursive,
+            [
+                'params.php',
```

Comment:
Is it necessary to specify `.php`? What if my `params` is mapped to `paramz.php` in `composer.json`?

Reply:
It is not necessary.

Reply:
Let's remove it then.